kong/core/handler.lua

```diff
@@ -119,9 +121,47 @@ return {
       end
 
       -- if set `host_header` is the original header to be preserved
-      var.upstream_host = host_header or 
+      var.upstream_host = host_header or
           balancer_address.hostname..":"..balancer_address.port
 
+      -- Keep-Alive and WebSocket Protocol Upgrade Headers
+      if var.http_upgrade == "websocket" then
```

Comment:
While I was doing it, I thought, why not do them all similarly in Lua-land (they are somewhat related). But that case-insensitivity thing was something I didn't think about. Great catch! Basically two choices, case insensitive match on Lua or move them back to maps.